spec/support/features/session_helper.rb

```diff
@@ -50,6 +50,7 @@ def sign_in_user(user = create(:user))
     end
 
     def sign_in_before_2fa(user = create(:user))
+      allow(FeatureManagement).to receive(:prefill_otp_codes?).and_return(true)
```

Comment:
We are adding this here to allow the tests on lines 47 and 81 and to pass. Otherwise, submitting the OTP form would be submitting a blank OTP, which would be invalid.